hudi-common/src/main/java/org/apache/hudi/common/table/view/RemoteHoodieTableFileSystemView.java

```diff
@@ -165,17 +176,17 @@ private <T> T executeRequest(String requestPath, Map<String, String> queryParame
 
     String url = builder.toString();
     LOG.info("Sending request : (" + url + ")");
-    Response response;
     int timeout = this.timeoutSecs * 1000; // msec
-    switch (method) {
-      case GET:
-        response = Request.Get(url).connectTimeout(timeout).socketTimeout(timeout).execute();
-        break;
-      case POST:
-      default:
-        response = Request.Post(url).connectTimeout(timeout).socketTimeout(timeout).execute();
-        break;
-    }
+    RetryHelper.CheckedFunction<Response> checkedFun = () -> {
+      switch (method) {
+        case GET:
+          return Request.Get(url).connectTimeout(timeout).socketTimeout(timeout).execute();
+        case POST:
+        default:
+          return Request.Post(url).connectTimeout(timeout).socketTimeout(timeout).execute();
```

Comment:
@nsivabalan , can you help with this part if you have time, the code is not efficient compared to the original, wondering if you have some good idea here ?

Reply:
Can we add a parameter `url` to `RetryHelper.CheckedFunction`, In this way, each time call only the `url` attribute of `CheckedFunction` needs to be modified.

Reply:
Hi, @nsivabalan , can you help to review this PR if you have time